### PR TITLE
replace travis script to enable caching

### PR DESCRIPTION
**Pull request by jberdine**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,26 @@
...
+  - chmod +x ${HOME}/opam
+  - export PATH=${HOME}:${PATH}
+  - export OPAMYES=1
+  - export OPAMJOBS=2
+  - opam init --compiler=${OCAML_VERSION} --disable-sandboxing
+  - eval $(opam env)
+  - export OPAMVERBOSE=1
+install:
+  - opam update --upgrade
+  - opam pin add --no-action --working-dir infer .
```
Comment (jvillard) [inline .travis.yml]: Why `--working-dir`?

File: .travis.yml [hunk 2]
```diff
@@ -1,9 +1,26 @@
...
+  - opam depext -u infer
+  - opam install --deps-only --working-dir infer
+script:
+  - opam install --working-dir infer
 os:
 - linux
-- osx
```
Comment (jvillard) [inline .travis.yml]: :+1:

File: .travis.yml [hunk 3]
```diff
@@ -1,9 +1,26 @@
 language: c
 sudo: required
-install: wget https://raw.githubusercontent.com/ocaml/ocaml-ci-scripts/master/.travis-opam.sh
-script: bash -ex .travis-opam.sh
 env:
-  - OCAML_VERSION=4.07 TESTS=false
+  - OCAML_VERSION=4.07.1 TESTS=false
```
[the remark this answers is not in the document]
Comment (jberdine) [reply .travis.yml]: I didn't know if it was needed, will remove.

File: .travis.yml [hunk 4]
```diff
@@ -1,9 +1,26 @@
...
+cache:
+  directories:
+    - ${HOME}/.opam
+    - facebook-clang-plugins
+before_install:
+  - wget -O ${HOME}/opam https://github.com/ocaml/opam/releases/download/2.0.3/opam-2.0.3-x86_64-linux
+  - chmod +x ${HOME}/opam
+  - export PATH=${HOME}:${PATH}
+  - export OPAMYES=1
+  - export OPAMJOBS=2
```
[the remark this answers is not in the document]
Comment (jberdine) [reply .travis.yml]: It seems to be a bit faster and is a common value. The vms are single core, so leaving it unset is sequential.

File: .travis.yml (hunk 1 above)
Comment (jberdine) [reply .travis.yml]: It's a failed attempt to prevent opam from uselessly making a copy of the whole dir, will remove.

**Pull request closed without merging**